### PR TITLE
GFORMS-2519: capture auth wizzard data

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

**Review by josef-vlach — CHANGES_REQUESTED** [review]
Couple of minor comments

File: app/uk/gov/hmrc/gform/testonly/AuthLoginStubService.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      e => throw new Exception("can not authenticate user"),
+      identity
+    )
+
+  private def login(
+    loginData: GovernmentGatewayFormData
+  ): EitherT[Future, UnexpectedState, Session] =
+    connector.login(loginData).subflatMap { httpResponse =>
+      val status = httpResponse.status
+      lazy val location = httpResponse.header("location")
```
[the remark this answers is not in the document]
Comment (lxol) [reply app/uk/gov/hmrc/gform/testonly/AuthLoginStubService.scala:49]: :thinking:  removed it

File: app/uk/gov/hmrc/gform/testonly/AuthLoginStubService.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  connector: AuthLoginStubConnector,
+  sessionCookieCrypto: SessionCookieCrypto,
+  sessionCookieBaker: SessionCookieBaker
+)(implicit ec: ExecutionContext) {
+
+  def getSession(
+    loginData: GovernmentGatewayFormData
+  ): Future[Session] =
+    login(loginData).fold(
+      e => throw new Exception("can not authenticate user"),
```
[the remark this answers is not in the document]
Comment (lxol) [reply app/uk/gov/hmrc/gform/testonly/AuthLoginStubService.scala:40]: added `e` to to the message

File: app/uk/gov/hmrc/gform/testonly/AuthStubModel.scala [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  def withEmail(email: Option[String]): GovernmentGatewayFormData =
+    copy(email = email)
+
+  def withGatewayToken(gatewayInformation: Option[GatewayInformation]): GovernmentGatewayFormData =
+    copy(gatewayToken = gatewayInformation.flatMap(_.gatewayToken))
+
+  def withGroupIdentifier(groupIdentifier: Option[String]): GovernmentGatewayFormData =
+    copy(groupIdentifier = groupIdentifier)
+  import java.time.LocalDate
+  import java.time.format.DateTimeFormatter
```
[the remark this answers is not in the document]
Comment (lxol) [reply app/uk/gov/hmrc/gform/testonly/AuthStubModel.scala]: @josef-vlach I updated the PR

**Pull request merged**